### PR TITLE
Server commands2

### PR DESCRIPTION
**Pull request by openbots-nc**
- Add SendServerEmailCommand to list of Commands
- Refactor QueueItem commands to allow for attachment of files
- Test ServerEmail, QueueItem, Credential, and Asset commands to ensure proper functionality

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.QueueItem/WorkQueueItemCommand.cs [hunk 1]
```diff
@@ -88,9 +113,24 @@ public override void RunCommand(object sender)
 													   kvp.Key == "DataJson" ||
 													   kvp.Key == "Priority" ||
 													   kvp.Key == "LockedUntilUTC")
-										 .ToDictionary(i =>i.Key, i => i.Value);
+										 .ToDictionary(i => i.Key, i => i.Value);
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.QueueItem/WorkQueueItemCommand.cs:116]: Do queueitems have a property for the number of attachments? If they do, you might want to add that here. What this is is a filter of just the relevant properties that the user would want to see in studio debug.

File: OpenBots.Commands/OpenBots.Commands.ServerEmail/OpenBots.Commands.ServerEmail.nuspec [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <version>1.2.0.0</version>
+    <title>ServerEmail Commands</title>
+    <authors>OpenBots Inc</authors>
+    <owners>OpenBots Inc</owners>
+    <licenseUrl>https://github.com/OpenBotsAI/OpenBots.Studio/blob/develop/LICENSE</licenseUrl>
+    <projectUrl>https://openbots.ai</projectUrl>
+    <iconUrl>https://openbots.ai/wp-content/uploads/2020/11/Studio-Icon-256px.png</iconUrl>
+    <repository type="git" url="https://github.com/OpenBotsAI/OpenBots.Studio" branch="master" />
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>OpenBots command for sending an email message with optional attachment(s) in OpenBots Server.</description>
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.ServerEmail/OpenBots.Commands.ServerEmail.nuspec:14]: This description and the one in AssemblyInfo.cs should match

**Review by openbots-ff — CHANGES_REQUESTED** [review]

**Review by openbots-ff — CHANGES_REQUESTED** [review]
Please update the UIImage.cs file as well

File: OpenBots.Commands/OpenBots.Commands.QueueItem/WorkQueueItemCommand.cs (hunk 1 above)
Comment (openbots-nc) [reply OpenBots.Commands/OpenBots.Commands.QueueItem/WorkQueueItemCommand.cs:116]: No but I have to refactor queue items for the next version.  I like that suggestion though and will add it to my to do list.

File: OpenBots.Commands/OpenBots.Commands.ServerEmail/OpenBots.Commands.ServerEmail.nuspec (hunk 2 above)
Comment (openbots-nc) [reply OpenBots.Commands/OpenBots.Commands.ServerEmail/OpenBots.Commands.ServerEmail.nuspec:14]: Done

**Review by openbots-ff — DISMISSED** [review]

File: OpenBots.Commands/OpenBots.Commands.Asset.Tests/UpdateAssetCommandTests.cs [hunk 3]
```diff
@@ -131,9 +131,12 @@ public void UpdatesFileAsset()
             _getAsset.v_AssetType = "File";
             _getAsset.v_OutputDirectoryPath = filepath + @"Download\";
 
+            if (!Directory.Exists(_getAsset.v_OutputDirectoryPath))
+                Directory.CreateDirectory(_getAsset.v_OutputDirectoryPath);
+
             _getAsset.RunCommand(_engine);
 
-            string outputAsset = "{output}".ConvertUserVariableToString(_engine);
+            //string outputAsset = "{output}".ConvertUserVariableToString(_engine);
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Asset.Tests/UpdateAssetCommandTests.cs:139]: do you still need this comment?

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.Asset.Tests/UpdateAssetCommandTests.cs (hunk 3 above)
Comment (openbots-nc) [reply OpenBots.Commands/OpenBots.Commands.Asset.Tests/UpdateAssetCommandTests.cs:139]: No I'll remove it now

**Review by openbots-ff — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**